### PR TITLE
Added new endpoint umls-auth to validate umls key.

### PR DESCRIPTION
**Pull request by DerekFurstPitt**
Added new endpoint umls-auth to validate umls key. 

Added accompanying config values to app.cfg.example

### CODE REVIEW

File: hubmap-auth/src/app.py [hunk 1]
```diff
@@ -284,6 +284,47 @@ def file_auth():
...
+    response_200 = make_response(jsonify({"message": "OK: Authorized"}), 200)
+    response_401 = make_response(jsonify({"message": "ERROR: Unauthorized"}), 401)
+    response_403 = make_response(jsonify({"message": "ERROR: Forbidden"}), 403)
+    response_500 = make_response(jsonify({"message": "ERROR: Internal Server Error"}), 500)
+
+    # Note: 400 and 404 are not supported http://nginx.org/en/docs/http/ngx_http_auth_request_module.html
+    # Any response code other than 200/401/403 returned by the subrequest is considered an error 500
+    # The end user or client will never see 404 but 500
+    response_400 = make_response(jsonify({"message": "ERROR: Bad Request"}), 400)
+    response_404 = make_response(jsonify({"message": "ERROR: Not Found"}), 404)
```
Comment (yuanzhou) [inline hubmap-auth/src/app.py:304]: We should remove the use of 400 and 404 here, will never really happen. And the users will only get 500. 404 can happen when the auth passes and the requested file doesn't exist on file system, but not from here.

File: hubmap-auth/src/app.py [hunk 2]
```diff
@@ -284,6 +284,47 @@ def file_auth():
...
+        return response_400
+    parsed_uri = urlparse(orig_uri)
+
+    logger.debug("======parsed_uri======")
+    logger.debug(parsed_uri)
+
+    query = parse_qs(parsed_uri.query)
+
+    if 'umls-key' not in query:
+        return response_400
```
Comment (yuanzhou) [inline hubmap-auth/src/app.py:321]: When the UMLS key is not provided in the URL, we should consider this as 401. Plus returning 400 will become 500.

File: hubmap-auth/src/app.py [hunk 3]
```diff
@@ -284,6 +284,47 @@ def file_auth():
...
+    response_400 = make_response(jsonify({"message": "ERROR: Bad Request"}), 400)
+    response_404 = make_response(jsonify({"message": "ERROR: Not Found"}), 404)
+
+    orig_uri = None
+
+    if "X-Original-URI" in request.headers:
+        orig_uri = request.headers.get("X-Original-URI")
+
+    if orig_uri is None:
+        return response_400
```
Comment (yuanzhou) [inline hubmap-auth/src/app.py:312]: As long as the sub-request gets to this `/umls_auth` endpoint, this `orig_uri` won't be `None`. So no need to check here, plus returning 400 will become 500 and cause confusions.

**Review by yuanzhou — CHANGES_REQUESTED** [review]

**Comment by DerekFurstPitt** [comment]
@yuanzhou requested changes applied

**Comment by yuanzhou** [comment]
I had tested on DEV, will go to `main` directly since we'll only deploy to PROD for now.

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**